tests/test_docker-compose.sh

```diff
@@ -0,0 +1,660 @@
+        return $rc
+    fi
+
+    # Now we need to change the file tree to be a new artifact
+    rm -rf "${WORKDIR}/artifact/tmp/"*
+    cat << EOF > "${WORKDIR}/artifact/header/header-info"
+{
+  "artifact_provides": { "artifact_name": "test-artifact2" }
+}
+EOF
+
+    rm "${WORKDIR}/images/image2.tar"
+    tar -C "${WORKDIR}" -czf "${WORKDIR}/artifact/files/images.tar.gz" images
+    cat << EOF > "${WORKDIR}/manifests/docker-compose.yml"
+services:
+  lighttpd:
+    image: some/lighttpd:latest
+    ports:
+      - "8081:80"
+EOF
+    tar -C "${WORKDIR}" -cf "${WORKDIR}/artifact/files/manifests.tar" manifests
+
+    "${SRCDIR}/docker-compose" ArtifactInstall "${WORKDIR}/artifact" > "${WORKDIR}/docker-compose.log" 2>&1 || rc=$?
+    if [ $rc -ne 0 ]; then
+        echo "Second ArtifactInstall failed (exit code $rc), logs follow:"
+        cat "${WORKDIR}/docker-compose.log"
+        return $rc
+    fi
+    "${SRCDIR}/docker-compose" ArtifactCommit "${WORKDIR}/artifact" >> "${WORKDIR}/docker-compose.log" 2>&1 || rc=$?
+    if [ $rc -ne 0 ]; then
+        echo "Second ArtifactCommit failed (exit code $rc), logs follow:"
+        cat "${WORKDIR}/docker-compose.log"
+        return $rc
+    fi
+
+    "${SRCDIR}/docker-compose" Cleanup "${WORKDIR}/artifact" >> "${WORKDIR}/docker-compose.log" 2>&1 || rc=$?
+    if [ $rc -ne 0 ]; then
+        echo "Second Cleanup failed (exit code $rc), logs follow:"
+        cat "${WORKDIR}/docker-compose.log"
+        return $rc
+    fi
+
+    cat << EOF | diff -u - "$CMDLINE_LOGGER_LOG_FILE" || rc=$?
+docker image load
+docker image load
+docker images --format {{json .ID}} some/lighttpd:latest
+docker images --format {{json .ID}} bad/php:oldest
+docker-compose --project-name test-comp up -d
+docker image load
+docker images --format {{json .ID}} some/lighttpd:latest
+docker-compose --project-name test-comp down
+docker-compose --project-name test-comp up -d
+docker rmi bad/php:oldest
+EOF
+    if [ $rc -ne 0 ]; then
+        echo "Unexpected commands executed (see the above diff), logs follow:"
+        cat "${WORKDIR}/docker-compose.log"
+        return $rc
+    fi
+
+    if [ -d "${PERSISTENT_DIR}/new" ]; then
+        echo "'new' directory exists after second commit"
+        return 1
+    fi
+    if ! [ -d "${PERSISTENT_DIR}/current" ]; then
+        echo "'current' directory doesn't exist after second commit"
+        return 1
+    fi
+    if [ -d "${PERSISTENT_DIR}/cleanup" ]; then
+        echo "'cleanup' directory exists after second commit"
+        return 1
+    fi
+
+    return $rc
+}
+tests+=(test_artifact_install_commit_install_commit_cleanup)
+
+test_artifact_install_commit_install_rollback_cleanup() {
+    local rc=0
+
+    prepare_config
+    prepare_expected_file_tree
+    prepare_docker_mock
+
+    # For this test, we need 'docker images...' to return something consistent
+    # between runs that we can easily refer to.
+    cat << EOF > "${WORKDIR}/bin/docker"
```

Comment:
This mock could have been moved to a shared function between these two tests that required the previous artifact.